### ISSUE TITLE
Accept stdin as input

### ISSUE THREAD

**Issue by har-wradim** [scapella/trimal#24]
A feature request: would be very convenient to have an option to pipe into trimal:

    my_pipe | trimal -in /dev/stdin

or

    my_pipe | trimal -in -

**Comment by scapella** [comment]
Good idea - It shouldn't be difficult to implement it in the coming days.
Could you, please, help me to test it once implement it?

**Comment by har-wradim** [comment]
Sure!

**Comment by scapella** [comment]
It is going to take a bit longer than expected because I need to figure out
how to read twice the input data.
Although inefficient it is the only way to determine which format is used
for the input alignment when it is not explicitly specified.

S


On Mon, Nov 9, 2015 at 12:33 AM, Andrey Rozenberg <notifications@github.com>
wrote:

> Sure!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scapella/trimal/issues/24#issuecomment-154886661>.
>



-- 
Salva

**Comment by har-wradim** [comment]
I have just discovered the -backtrans option. This is great (!), but the bottleneck is that -backtrans has the same behavior as -in in that it doesn't allow piping. An exemplary workflow would ideally look like:

    trimal \
        -in        <(getAAfasta | mafft --amino) \
        -backtrans <(getCDfasta)                 \
    > output

(or some syntactic variation thereof), where getAAfasta and getCDfasta are some piping function with actual sequences coming from a database or some other pipes.

**Comment by fgvieira** [comment]
@scapella did you end up implementing this?

If not, I guess you can just add an argument where the user can specify the input format, and make it mandatory if reading input from the stdin.

Either way, if this options is specified, then trimal does not try to infer input file format.
